src/data_structures/weightedg.py

```diff
@@ -0,0 +1,46 @@
+# coding=utf-8
+from data_structures.simpleg import Graph
+
+
+class WeightedGraph(Graph):
+    def add_node(self, node):
+        self._dict.setdefault(node, {})
+
+    def del_node(self, n):
+        """
+        delete the node ‘n’ from the graph, raises an error if no such node
+        exists
+        """
+        # remove the node
+        self._dict.pop(n)
+        # remove any references to n
+        for node in self._dict:
+            self._dict[node].pop(n)
+
+    def add_edge(self, start, end, weight=1):
+        """
+        add a new edge to the graph connecting ‘n1’ and ‘n2’, if either n1
+        or n2 are not already present in the graph, they should be added.
+        """
+        self.add_node(start)
+        self.add_node(end)
+        self._dict[start][end] = weight
```

Comment:
You should enforce that your weight cannot be a string. Preferably it'd be numerical, or some container that holds multiple weights for multiple relationships.